pkg/messaging/topics.go

```diff
@@ -154,3 +137,70 @@ func EncodeMessageMQTTTopic(m *Message) string {
 	}
 	return topic
 }
+
+// ParseTopic parses a messaging topic string and returns the domain ID, channel ID, and subtopic.
+// This is an optimized version with no regex and minimal allocations.
+func ParseTopic(topic string) (domainID, chanID, subtopic string, err error) {
+	// location of string "m"
+	start := 0
+	// Handle both formats: "/m/domain/c/channel/subtopic" and "m/domain/c/channel/subtopic".
+	// If topic start with m/ then start is 0 , If topic start with /m/ then start is 1.
+	n := len(topic)
+	if n > 0 && topic[0] == '/' {
+		start = 1
+	}
+
+	// length check - minimum: "m/<domain_id>/c/" = 5 characters if ignore <domain_id> and in this case start will be 0
+	// length check - minimum: "/m/<domain_id>/c/" = 6 characters if ignore <domain_id> and in this case start will be 1
+	if n < start+5 {
+		return "", "", "", ErrMalformedTopic
+	}
+	if topic[start] != 'm' || topic[start+1] != '/' {
```

Comment:
We can use constant for Prefix chars like below
```suggestion
	if topic[start] != MsgTopicPrefix || topic[start+1] != '/' {
```
But this requires following changes 

```go 
const SubjectAllMessages = string(messaging.MsgTopicPrefix) + ".>"

const (
	MsgTopicPrefix     = 'm'
	ChannelTopicPrefix = 'c'
)

```

Reply:
In the file we have already defined:
![image](https://github.com/user-attachments/assets/7545672a-3e97-46f3-b913-9285ecc5d9d7)

and they are strings and not runes, so we cant use them in the above case

Reply:
yeah , So only i suggested to change them to rune like below 

```
const (  
    MsgTopicPrefix     = 'm'  
    ChannelTopicPrefix = 'c'  
)  

```

and all other used palace like below, but there only one place it is used 
```
const SubjectAllMessages = string(messaging.MsgTopicPrefix) + ".>"  
```